kzg_prover/src/chips/range/range_check.rs

```diff
@@ -1,55 +1,59 @@
 /// * `range`: Fixed column for the lookup table. It contains the values from 0 to 2^8 - 1.
-/// * `lookup_enable_selector`: Selector to enable the lookup check.
 ///
 /// Patterned after [halo2_gadgets](https://github.com/privacy-scaling-explorations/halo2/blob/main/halo2_gadgets/src/utilities/decompose_running_sum.rs)
 #[derive(Debug, Copy, Clone)]
 pub struct RangeCheckConfig<const N_BYTES: usize> {
-    z: Column<Advice>,
-    range: Column<Fixed>,
-    lookup_enable_selector: Selector,
+    zs: [Column<Advice>; N_BYTES],
 }
 
-/// Helper chip that verfiies that the value witnessed in a given cell lies within a given range defined by N_BYTES.
+/// Helper chip that verfiies that the element witnessed in a given cell lies within a given range defined by N_BYTES.
 /// For example, Let's say we want to constraint 0x1f2f3f4f to be within the range N_BYTES=4.
 ///
-/// `z(0) = 0x1f2f3f4f`
-/// `z(1) = (0x1f2f3f4f - 0x4f) / 2^8 = 0x1f2f3f`
-/// `z(2) = (0x1f2f3f - 0x3f) / 2^8 = 0x1f2f`
-/// `z(3) = (0x1f2f - 0x2f) / 2^8 = 0x1f`
-/// `z(4) = (0x1f - 0x1f) / 2^8 = 0x00`
+/// `z = 0x1f2f3f4f`
+/// `zs[0] = (0x1f2f3f4f - 0x4f) / 2^8 = 0x1f2f3f`
+/// `zs[1] = (0x1f2f3f - 0x3f) / 2^8 = 0x1f2f`
+/// `zs[2] = (0x1f2f - 0x2f) / 2^8 = 0x1f`
+/// `zs[3] = (0x1f - 0x1f) / 2^8 = 0x00`
 ///
-///  |     | z          |
-///  | --- | ---------- |
-///  | 0   | 0x1f2f3f4f |
-///  | 1   | 0x1f2f3f   |
-///  | 2   | 0x1f2f     |
-///  | 3   | 0x1f       |
-///  | 4   | 0x00       |
+///  z          | zs[0]      | zs[1]      | zs[2]      | zs[3]      |
+///  ---------  | ---------- | ---------- | ---------- | ---------- |
+///  0x1f2f3f4f | 0x1f2f3f   | 0x1f2f     | 0x1f       | 0x00       |
+///
+/// Column zs[0], at offset 0, contains the truncated right-shifted value z - ks[0] / 2^8 (shift right by 8 bits) where ks[0] is the 0-th decomposition big-endian of the element to be checked
+/// Column zs[1], at offset 0, contains the truncated right-shifted value zs[0] - ks[1] / 2^8 (shift right by 8 bits) where ks[1] is the 1-th decomposition big-endian of the element to be checked
+/// Column zs[2], at offset 0, contains the truncated right-shifted value zs[1] - ks[2] / 2^8 (shift right by 8 bits) where ks[2] is the 2-th decomposition big-endian of the element to be checked
+/// Column zs[3], at offset 0, contains the truncated right-shifted value zs[2] - ks[3] / 2^8 (shift right by 8 bits) where ks[3] is the 3-th decomposition big-endian of the element to be checked
 ///
-/// The column z contains the witnessed value to be checked at offset 0
-/// At offset i, the column z contains the value z(i+1) = (z(i) - k(i)) / 2^8 (shift right by 8 bits) where k(i) is the i-th decomposition big-endian of `value`
 /// The contraints that are enforced are:
-/// - z(i) - 2^8⋅z(i+1) ∈ lookup_u8 (enabled by lookup_enable_selector at offset [0, N_BYTES - 1])
-/// - z(N_BYTES) == 0
+/// 1.
+/// - z - 2^8⋅zs[0] = kz[0] ∈ lookup_u8
```

Comment:
`z` and `zs[i]` are not negative in the constraints but specified as negative in the description.

Reply:
Lol yes that is a bullet point. Sorry for that. Gonna fix it